cmd/terramate/cli/run.go

```diff
@@ -544,6 +561,7 @@ func (c *cli) runAll(
 			if err != nil {
 				c.cloudSyncAfter(cloudRun, runResult{ExitCode: -1}, errors.E(ErrRunCommandNotFound, err))
 				errs.Append(errors.E(err, "running `%s` in stack %s", cmdStr, run.Stack.Dir))
+				releaseResource()
```

Comment:
This line fixes the deadlock reproduced in the `TestParalleCmdNotFoundContinueOnError` testcase.